test/new-e2e/tests/agent-platform/install-script/install_script_test.go

```diff
@@ -74,36 +75,78 @@ func TestInstallScript(t *testing.T) {
-	unixHelper := helpers.NewUnixHelper()
+
 	vm := is.Env().VM.(*client.PulumiStackVM)
 	agentClient, err := client.NewAgentClient(is.T(), vm, vm.GetOS(), false)
 	require.NoError(is.T(), err)
-	client := common.NewTestClient(is.Env().VM, agentClient, fileManager, unixHelper)
-
-	install.Unix(is.T(), client, installparams.WithArch(*architecture))
-
-	common.CheckInstallation(is.T(), client)
-	common.CheckAgentBehaviour(is.T(), client)
-	common.CheckAgentStops(is.T(), client)
-	common.CheckAgentRestarts(is.T(), client)
-	common.CheckIntegrationInstall(is.T(), client)
-	common.CheckAgentPython(is.T(), client, "3")
-	common.CheckApmEnabled(is.T(), client)
-	common.CheckApmDisabled(is.T(), client)
-	if is.cwsSupported {
-		common.CheckCWSBehaviour(is.T(), client)
+
+	// Tests for datadog-agent flavor
+	if *flavor == "datadog-agent" || *flavor == "datadog-heroku-agent" {
+		unixHelper := helpers.NewUnixHelper()
+		client := common.NewTestClient(is.Env().VM, agentClient, fileManager, unixHelper)
+
+		install.Unix(is.T(), client, installparams.WithArch(*architecture), installparams.WithFlavor(*flavor))
+
+		common.CheckInstallation(is.T(), client)
+		common.CheckAgentBehaviour(is.T(), client)
+		common.CheckAgentStops(is.T(), client)
+		common.CheckAgentRestarts(is.T(), client)
+		common.CheckIntegrationInstall(is.T(), client)
+		common.CheckAgentPython(is.T(), client, "3")
+		common.CheckApmEnabled(is.T(), client)
+		common.CheckApmDisabled(is.T(), client)
+		if is.cwsSupported {
+			common.CheckCWSBehaviour(is.T(), client)
+		}
+		common.CheckInstallationInstallScript(is.T(), client)
+		common.CheckUninstallation(is.T(), client, "datadog-agent")
+		return
+	}
+
+	// Tests for datadog-iot-agent flavor
+	if *flavor == "datadog-iot-agent" {
+
+		unixHelper := helpers.NewUnixHelper()
+		client := common.NewTestClient(is.Env().VM, agentClient, fileManager, unixHelper)
+
+		install.Unix(is.T(), client, installparams.WithArch(*architecture), installparams.WithFlavor(*flavor))
+
+		common.CheckInstallation(is.T(), client)
+		common.CheckAgentBehaviour(is.T(), client)
+		common.CheckAgentStops(is.T(), client)
+		common.CheckAgentRestarts(is.T(), client)
+
+		common.CheckInstallationInstallScript(is.T(), client)
+		common.CheckUninstallation(is.T(), client, "datadog-iot-agent")
+		return
+	}
+
+	if *flavor == "datadog-dogstatsd" {
+
+		unixHelper := helpers.NewUnixDogstatsdHelper()
+		client := common.NewTestClient(is.Env().VM, agentClient, fileManager, unixHelper)
+
+		install.Unix(is.T(), client, installparams.WithArch(*architecture), installparams.WithFlavor(*flavor))
+
+		common.CheckInstallation(is.T(), client)
+		common.CheckDogstatdAgentBehaviour(is.T(), client)
+		common.CheckDogstatsdAgentStops(is.T(), client)
+		common.CheckDogstatsdAgentRestarts(is.T(), client)
+		common.CheckInstallationInstallScript(is.T(), client)
+		common.CheckUninstallation(is.T(), client, "datadog-dogstatsd")
+
+		return
+
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You might extract 3 helpers for each flavor, and call them from a switch

Reply:
Created three function and called them in a switch. I think the code is better like that, thanks!